dj/api/nodes.py

```diff
@@ -272,6 +275,58 @@ def delete_a_node(name: str, *, session: Session = Depends(get_session)):
     return Response(status_code=HTTPStatus.NO_CONTENT.value)
 
 
+def cube_materialization_config(cube_node: NodeRevision, config: Dict):
+    """
+    Builds the materialization config for a cube. This includes two parts:
+    (a) building a query that decomposes each of the cube's metrics into their
+    constituent measures that have simple aggregations
+    (b) adding a metric to measures mapping that tells us which measures
+    in the query map to each selected metric
+
+    The query in the materialization config is different from the one stored
+    on the cube node itself in that this one is meant to create a temporary
+    table in preparation for ingestion into an OLAP database like Druid. The
+    metric to measures mapping then provides information on how to assemble
+    metrics from measures based on this query's output.
+
+    The query directly on the cube node is meant for direct querying of the cube
+    without materialization to an OLAP database.
+    """
+    combined_ast = parse(cube_node.query)
+    dimensions_set = {
+        dim.name for dim in cube_node.columns if dim.has_dimension_attribute()
+    }
+    metrics_to_measures = decompose_metrics(combined_ast, dimensions_set)
```

Comment:
I feel that if anyone is ever confused about the difference between measures and metrics, this line makes it as clear as it gets :)